app/models/file_depot_s3.rb

```diff
@@ -13,8 +13,6 @@ def connect(options = {})
     require 'aws-sdk'
     username = options[:username] || authentication_userid(options[:auth_type])
     password = options[:password] || authentication_password(options[:auth_type])
-    # Note: The hard-coded aws_region will be removed after manageiq-ui-class implements region selection
-    aws_region = options[:region] || "us-east-1"
```

Comment:
This will fail as is. `aws_region` is being used below but not set. Can you add a spec to test this method which should catch simple mistakes like this?

Reply:
I hate to disagree @carbonin but it works and has been tested.  The value is set in MiqSchedule#verify_file_depot.

Reply:
Ah forgot that there is an attribute called `aws_region`. The local variable was shadowing that I guess.

If there were specs then I wouldn't need to worry about interpreting the code to ensure it works though. Please add some to test for this.

Reply:
@carbonin I'm going back to look at this test you requested.  I want to explicitly call out what I believe this spec test would do --  You're asking that if I create the object and set the aws_region, then call connect - that the aws_region is still set.  Am I right?  I can think of no way to test this besides  recording a cassette of a real connection to AWS, or creating a mock class.  And I'm not really sure that this test is something that is worth that effort.  Please feel free to disagree.  :-)  Also - let me ask - do we typically set an attribute on an object in a spec, and then test to see if it is set?

Reply:
I'm asking for you to add a test to ensure that the correct arguments are passed to `Aws::S3::Resource.new` given which instance variables are set and the contents of the options hash passed to this method.

Ideally there would have been a spec that tested that the `:region` key for the call to `Aws::S3::Resource.new` would be defaulted if none was passed in the options hash. And another which tested that the correct region was passed when one was set in the options hash.

In this new case you don't have to pass a region in options (because that behavior is being removed), but I would still test that the call to `Aws::S3::Resource.new` uses the region from the instance variable.

Additionally, there should be specs that test that you can override the username and password using the options hash and that if none is passed they are fetched from the authentications table.